packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -0,0 +1,945 @@
+///
+/// ## ToggleButton Borders
+/// The toggle buttons, by default, have a solid, 1 dp border surrounding itself
+/// and separating each button. The toggle button borders' color, width, and
+/// corner radii are configurable.
+///
+/// The [selectedBorderColor] determines the border's color when the button is
+/// selected, while [disabledBorderColor] determines the border's color when
+/// the button is disabled. [borderColor] is used when the button is enabled.
+///
+/// To remove the border, set [borderWidth] to null. Setting [borderWidth] to
+/// 0.0 results in a hairline border. For more information on hairline borders,
+/// see [BorderSide.width].
+///
+/// See also:
+///
+///  * <https://material.io/design/components/buttons.html#toggle-button>
+class ToggleButtons extends StatelessWidget {
+  /// Creates a horizontal set of toggle buttons.
+  ///
+  /// It displays its widgets provided in a [List] of [children] horizontally.
+  /// The state of each button is controlled by [isSelected], which is a list of
+  /// bools that determine if a button is in an active, disabled, or selected
+  /// state. They are both correlated by their index in the list.
+  ///
+  /// Both [children] and [isSelected] properties arguments are required.
+  const ToggleButtons({
+    Key key,
+    @required this.children,
+    @required this.isSelected,
+    this.onPressed,
+    this.color,
+    this.selectedColor,
+    this.disabledColor,
+    this.fillColor,
+    this.focusColor,
+    this.highlightColor,
+    this.hoverColor,
+    this.splashColor,
+    this.focusNodes,
+    this.renderBorder = true,
+    this.borderColor,
+    this.selectedBorderColor,
+    this.disabledBorderColor,
+    this.borderRadius,
+    this.borderWidth,
+  }) :
+    assert(children != null),
+    assert(isSelected != null),
+    assert(children.length == isSelected.length),
+    super(key: key);
+
+  /// The corresponding widget values in the toggle buttons.
+  ///
+  /// These are typically [Icon] or [Text] widgets. The selection state
+  /// corresponds to its state in the [isSelected] list.
+  ///
+  /// The length of children has to match the length of [isSelected]. If
+  /// [focusNodes] is not null, the length of children has to also match
+  /// the length of [focusNodes].
+  final List<Widget> children;
+
+  /// The corresponding selection state of each toggle button.
+  ///
+  /// The boolean values in the list map directly to [children] by its index.
+  /// The values in the list cannot be null.
+  ///
+  /// The length of [isSelected] has to match the length of [children].
+  final List<bool> isSelected;
+
+  /// The callback that is called when a button is tapped.
+  ///
+  /// When set to null, all toggle buttons will be disabled.
+  final void Function(int index) onPressed;
+
+  /// The color for [Text] and [Icon] widgets if the button is enabled.
+  ///
+  /// If [selected] is set to false and [onPressed] is not null, this color will be used.
+  final Color color;
```

Comment:
For some of the color properties here, it might be useful to allow for `MaterialStateProperty<Color>`s to be used. Making those changes would involve 

1. Keeping track of a `Set<MaterialState> _states` inside of `_ToggleButton`.
1. Calling `MaterialStateProperty.resolveAs<Color>(color, _states);` in the build method of `_ToggleButton`.
1. Specifying in the documentation here that if `color` is a `MaterialStateProperty<Color>`, `selectedColor`, and `disabledColor` will be ignored.

Reply:
This was something I have been thinking about as well, since we've discussed this offline before. Would it be something that we should add in a separate PR, or would it be better to just incorporate this now? 

cc/ @HansMuller

Reply:
We decided on creating a new PR and issue to track introducing `MaterialStateProperty<Color>`.  I think implementing it shouldn't be invasive and the current PR is already pretty big, so this would make the logic in this PR less complex.